### DISCUSSION TITLE
[Select] Dialog size

### DISCUSSION THREAD

**Discussion by aweell** [dxc-technology/halstack-react#901] [Ideas] [answered]
## The current problem

The select dialog width is tight to the trigger width changing with the `size` prop. When an option list doesn't fit the defined size we make an `overflow: ellipsis` (_Fig. 01_). Being the label of an option a necessary content to display in order to make a selection, this approach is leading to a bad user experience.

![select-ellipsis-example](https://user-images.githubusercontent.com/44420072/162436555-b348bc25-2f55-4eed-96fb-16fa3743882d.png)

_Fig. 01 - Select option ellipsis behavior_

## Proposed solutions

* Disconnect the dialog size from the trigger size and set the `width` to `max-content`
* Allow `line-break` in every dialog options

## Related issues
#900

**Comment by GomezIvann** [comment]
I'm continuing this discussion since we are receiving more and more requests to allow option labels to be fully displayed. I think the solution depends on the component:

- The most common anatomy of a Dropdown is a button that triggers a menu. The menu size does not necessarily need to be the same as the button since, in most cases, buttons have a `fitContent` width with a small label, while the option labels in the dropdown menu are always a bit longer. I don't see any direct relationship between the trigger and the menu.
In this case, I would go for a menu whose width is not related to the trigger, but to the longest label of its options.

![image](https://user-images.githubusercontent.com/44321109/220587634-69f14a03-bf02-45eb-ac48-9d93ca269a90.png)
_Dropdown menu anatomy, from [the component gallery](https://component.gallery/components/dropdown-menu/)_

- Regarding the Select, this is a bit more complex. I think we can all agree that the list should have the same size as the input that opens and closes it. So, my take here is to rethink a bit the spacing of each list option and allow line breaks for the labels and thus eliminate the current ellipsis. 

![image](https://user-images.githubusercontent.com/44321109/220604253-ab43d9c9-862d-485f-a22c-ee2aaa8dbb65.png)
_Example from [react-select](https://react-select.com/home)_

**Comment by GomezIvann** [comment, marked as answer]
After discussing with @rconcepcion3, we have decided we are gonna stay with the ellipsis to avoid possible redesigns when a long label has to coexist with a checkbox (multiple mode) and/or an icon. Instead, we are gonna add the built-in prop `title` to each option so it appears like this:

![image](https://user-images.githubusercontent.com/44321109/220947805-9a2d07f5-bf77-4615-886b-02c81d827140.png)

The built-in tooltip will display the full label until we implement our version of Tooltips.
